presto-hive/src/main/java/com/facebook/presto/hive/HiveClientConfig.java

```diff
@@ -133,6 +133,7 @@
     private boolean bucketExecutionEnabled = true;
     private boolean sortedWritingEnabled = true;
     private boolean ignoreTableBucketing;
+    private int maxBucketsForGroupedExecution = 1_500_000;
```

Comment:
Is this number decided by looking at how much files can be buffered with current default `256MB` buffer size?

Reply:
Yup